pkg/v1/remote/options.go

```diff
@@ -78,19 +107,21 @@ func makeOptions(target authn.Resource, opts ...Option) (*options, error) {
 		o.auth = auth
 	}
 
-	// Wrap the transport in something that logs requests and responses.
-	// It's expensive to generate the dumps, so skip it if we're writing
-	// to nothing.
-	if logs.Enabled(logs.Debug) {
-		o.transport = transport.NewLogger(o.transport)
-	}
+	if _, ok := o.transport.(*transport.Transport); !ok {
```

Comment:
Add a comment here and in `google` describing _why_ we're doing things this way.

Also for the `WithTransport` option, we'll want to describe this workaround.

Reply:
done!